src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -28,49 +32,68 @@ internal class SilentRequest : RequestBase
         {
             _silentParameters = silentParameters;
 
-            _brokerStrategyLazy = new Lazy<ISilentAuthRequestStrategy>(() => brokerStrategyOverride ?? new SilentBrokerAuthStrategy(this,
+            _brokerStrategyLazy = new Lazy<ISilentAuthRequestStrategy>(() => brokerStrategyOverride ?? new BrokerSilentStrategy(this,
                                                                                                serviceBundle,
                                                                                                authenticationRequestParameters,
                                                                                                silentParameters,
                                                                                                serviceBundle.PlatformProxy.CreateBroker(null)));
-            _clientStrategy = clientStrategyOverride ?? new SilentClientAuthStretegy(this, serviceBundle, authenticationRequestParameters, silentParameters);
+            _clientStrategy = clientStrategyOverride ?? new CacheSilentStrategy(this, serviceBundle, authenticationRequestParameters, silentParameters);
 
             _logger = authenticationRequestParameters.RequestContext.Logger;
         }
 
-        internal async override Task PreRunAsync()
-        {
-            await _clientStrategy.PreRunAsync().ConfigureAwait(false);
-        }
-
         protected override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
         {
+            await UpdateRequestWithAccountAsync().ConfigureAwait(false);
+            bool isBrokerConfigured =
+              AuthenticationRequestParameters.IsBrokerConfigured &&
+              ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth();
+
             try
             {
-                _logger.Info("Attempting to acquire token using using local cache...");
-                return await _clientStrategy.ExecuteAsync(cancellationToken).ConfigureAwait(false);
+                if (AuthenticationRequestParameters.Account != null)
+                {
+                    _logger.Verbose("Attempting to acquire token using local cache...");
+                    return await _clientStrategy.ExecuteAsync(cancellationToken).ConfigureAwait(false);
+                }
+
+                if (!isBrokerConfigured)
+                {
+                    _logger.Verbose("No account passed to AcquireTokenSilent and no broker configured.  ");
```

Comment:
>no broker configured [](start = 81, length = 20)

idk about adding broker messaging here. how many customers are using a broker at this point? I think that makes the actually error more confusing.

Reply:
It requires an aka.ms link;

Reply:
No, log messages do not require aka.ms links. This is verbose log message. The MSAL team (us) are the consumers of logs, not developers.

Reply:
I disagree on this one. Customers do consume the logs, as does css and anyone helping w/a support case or issue, so way more then the MSAL team. I would at least remove the "no broker configured" part.

Reply:
Ok no problem.